### PR TITLE
fix test, random order of slice cause test fail

### PR DESCRIPTION
**Pull request by suchen-sci**
fix TestDynamicService, random order of slice cause test fail

### CODE REVIEW

**Review by megaeasex — APPROVED** [review]
[TASK:easegress-pr-test SUCCESS]megaease/easegress Pull Request 224 Deploy Test Success

File: pkg/filter/proxy/server_test.go [hunk 1]
```diff
@@ -365,6 +365,9 @@ func TestDynamicService(t *testing.T) {
 	}
 
 	if !reflect.DeepEqual(wantStatic, s.static) {
-		t.Fatalf("want: %+v\ngot :%+v\n", wantStatic, s.static)
+		wantStatic.servers[0], wantStatic.servers[1] = wantStatic.servers[1], wantStatic.servers[0]
```
Comment (xxx7xxxx) [inline pkg/filter/proxy/server_test.go:368]: This solution is not general enough, what if it adds more servers. I think we should sort the `want` and `got` by URL then compare them.

Comment (suchen-sci) [reply pkg/filter/proxy/server_test.go:368]: good idea!

**Review by megaeasex — APPROVED** [review]
[TASK:easegress-pr-test SUCCESS]megaease/easegress Pull Request 224 Deploy Test Success

**Review by megaeasex — APPROVED** [review]
[TASK:easegress-pr-test SUCCESS]megaease/easegress Pull Request 224 Deploy Test Success

**Review by xxx7xxxx — APPROVED** [review]

**Review by localvar — APPROVED** [review]

**Pull request merged**